src/core/Tree/Tree.tsx

```diff
@@ -0,0 +1,228 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { CommonProps, useTheme } from '../utils';
+import '@itwin/itwinui-css/css/tree.css';
+import cx from 'classnames';
+
+export const TreeContext = React.createContext<
+  | {
+      selectedNodes?: Array<string>;
+      setSelectedNode?: (
+        node: Array<string> | ((prev: Array<string>) => Array<string>),
+      ) => void;
+      expandedNodes?: Array<string>;
+      setExpandedNodes?: (
+        node: Array<string> | ((prev: Array<string>) => Array<string>),
+      ) => void;
+      nodeDepth?: number;
+      parentNode?: FlatNode;
+    }
+  | undefined
+>(undefined);
+
+export type TreeData = {
```

Comment:
If we implement `getNode`, I am thinking we could change typings to this:
```suggestion
export type TreeData<T> = T & {
  subnodes?: Array<TreeData<T>>;
};

export type FlatNode<T> = {
  data: TreeData<T>;
  id: string;
  depth: number;
  subNodeIds: string[];
  parent?: FlatNode<T>; // still not sure about this
  isExpanded: boolean;
  isDisabled: boolean;
};
```

Reply:
Also, maybe rename to `NodeData` or `TreeNodeData`